lib/datadog/tracing/trace_operation.rb

```diff
@@ -101,6 +103,7 @@ def initialize(
         @trace_state = trace_state
         @trace_state_unknown_fields = trace_state_unknown_fields
         @tracer = tracer
+        @baggage = baggage || {}
```

Comment:
```suggestion
        @baggage = baggage
```

Can you avoid initializing the baggage with an empty hash? It's a good suggestion by @p-datadog!
You'll have to initialize it then in `tracing.rb` if it's not initialized.